packages/wallet/backend/tests/incomingPayment/controller.test.ts

```diff
@@ -0,0 +1,155 @@
+import { env } from '@/config/env'
+import {
+  createRequest,
+  createResponse,
+  MockRequest,
+  MockResponse
+} from 'node-mocks-http'
+import { NextFunction, Request, Response } from 'express'
+import { createContainer } from '@/createContainer'
+import { IncomingPaymentController } from '@/incomingPayment/controller'
+import {
+  mockIncomingPaymentGetPaymentDetailsByUrlRequest,
+  mockIncomingPaymentRequest,
+  mockIncomingPaymentRequestSession as mockIncomingPaymentRequestSessionUser,
+  mockIncomingPaymentService
+} from '../mocks'
+import { errorHandler } from '@/middleware/errorHandler'
+
+describe('Incoming Payment Controller', () => {
+  let req: MockRequest<Request>
+  let res: MockResponse<Response>
+  let incomingPaymentController: IncomingPaymentController
+
+  const next = jest.fn() as unknown as NextFunction
+
+  const createIncomingPaymentControllerDepsMock = () => {
+    const incomingPaymentControllerDepsMock = {
+      incomingPaymentService: mockIncomingPaymentService
+    }
+    Reflect.set(
+      incomingPaymentController,
+      'deps',
+      incomingPaymentControllerDepsMock
+    )
+  }
+
+  beforeAll(async () => {
+    const container = createContainer(env)
+    incomingPaymentController = await container.resolve(
+      'incomingPaymentController'
+    )
+    createIncomingPaymentControllerDepsMock()
+  })
+
+  beforeEach(async (): Promise<void> => {
+    const user = mockIncomingPaymentRequestSessionUser().user
+    req = createRequest({ session: { user } })
+    res = createResponse()
+  })
+
+  describe('Create', () => {
+    it('should call method create() in incomingPaymentService', async () => {
+      const createSpy = jest.spyOn(mockIncomingPaymentService, 'create')
+      req.body = mockIncomingPaymentRequest().body
+      await incomingPaymentController.create(req, res, next)
+      expect(createSpy).toHaveBeenCalled()
+      expect(createSpy).toHaveBeenCalledTimes(1)
+      expect(createSpy).toHaveBeenCalledWith(
+        req.session.user.id,
+        req.body.paymentPointerId,
+        req.body.amount,
+        req.body.description,
+        req.body.expiration
+      )
+    })
+    it("should return status 200 and message 'SUCCESS' if the payment was created", async () => {
```

Comment:
Please add new lines after every test function

Reply:
Of Course.

Reply:
@Tymmmy  it's allright now ?